aea/helpers/profiling.py

```diff
@@ -58,18 +58,14 @@ def get_current_process_cpu_time() -> float:
 
 else:
     import resource
+    import tracemalloc
 
     _MAC_MEM_STATS_MB = 1024 ** 2
     _LINUX_MEM_STATS_MB = 1024
 
     def get_current_process_memory_usage() -> float:
         """Get current process memory usage in MB."""
-        if platform.system() == "Darwin":  # pragma: nocover
-            divider = _MAC_MEM_STATS_MB
-        else:
-            divider = _LINUX_MEM_STATS_MB
-
-        return 1.0 * resource.getrusage(resource.RUSAGE_SELF).ru_maxrss / divider
+        return tracemalloc.get_traced_memory()[0] / 1024 ** 2
```

Comment:
return and report both `current` and `peak` memory usage.

Reply:
So peak would match (more or less) what we reported earlier?

If so, then obviously it would have to be returned elsewhere (in a new method) as this method is scoped to "current" by design and I don't want to unnecessarily break compatibility.